### PR TITLE
add e2es to group form

### PR DESCRIPTION
**Pull request by robguthrie**
privacy weirdness

I'm writing tests and fixing the behaviour of group form. This is the first part.

I'll add more to this PR as I go.

### CODE REVIEW

**Comment by robguthrie** [comment]
Errmm.. to be clear. This first commit causes the closed text to be constant. "Threads can be public or private."

File: angular/core/components/group_form/group_form.haml [hunk 1]
```diff
@@ -60,12 +60,12 @@
...
           %label{translate: "group_form.permissions"}
 
-          %label.lmo-form-labelled-input{ng-if: "group.privacyIsClosed() && !group.isSubgroupOfHiddenParent"}
+          %label.lmo-form-labelled-input{ng-if: "group.privacyIsClosed() && !group.isSubgroupOfSecretParent()"}
             %input.group-form__allow-public-threads{type: "checkbox", ng-model: "buh.allowPublicThreads", ng-change: "allowPublicThreadsClicked()"}
             %span{translate: "group_form.allow_public_threads"}
 
-          %label.lmo-form-labelled-input{ng-if: "group.isSubgroup() && !group.privacyIsSecret()"}
-            %input.group-form__visible-to-parent-members{type: "checkbox", ng-model: "group.parentMembersCanSeeDiscussions"}
+          %label.lmo-form-labelled-input{ng-if: "group.isSubgroup() && group.privacyIsClosed()"}
```
Comment (robguthrie) [inline angular/core/components/group_form/group_form.haml]: This option only makes sense for closed groups.

parentMembersCanSee(Private)Discussions

File: angular/core/components/start_menu/start_menu_option.haml [hunk 2]
```diff
@@ -1,5 +1,4 @@
-.start-menu__option
-  %a{href: "", ng-click: "openModal()"}
-    .start-menu__option-text{translate: "start_menu.{{text}}"}
-    .start-menu__option-icon.start-menu__menu-button
-      %i.fa.fa-lg{class: "fa-{{icon}}"}
+%button.start-menu__option{class: "start-menu__{{action}}", ng-click: "openModal()"}
```
Comment (robguthrie) [inline angular/core/components/start_menu/start_menu_option.haml]: These are now button elements. I reckon that is more correct.

File: angular/core/models/group_model.coffee [hunk 3]
```diff
@@ -121,7 +120,10 @@ angular.module('loomioApp').factory 'GroupModel', (DraftableModel, AppConfig) ->
       @groupPrivacy == 'secret'
 
     allowPublicDiscussions: ->
-      @discussionPrivacyOptions != 'private_only'
+      if @isNew()
```
Comment (robguthrie) [inline angular/core/models/group_model.coffee]: This prevents the text of Closed option changing when you select Secret

Comment (gdpelican) [inline angular/core/models/group_model.coffee]: Maybe this is a naming thing but I'm a little confused here; what happens when you make a new group and set it to 'secret'? Then allowPublicDiscussions would want to be false?

File: angular/core/models/group_model.coffee [hunk 4]
```diff
@@ -121,7 +120,10 @@ angular.module('loomioApp').factory 'GroupModel', (DraftableModel, AppConfig) ->
       @groupPrivacy == 'secret'
 
     allowPublicDiscussions: ->
-      @discussionPrivacyOptions != 'private_only'
+      if @privacyIsClosed() && @isNew()
```
Comment (robguthrie) [inline angular/core/models/group_model.coffee]: @gdpelican thanks for the comment .. is this clearer now?

File: angular/core/components/start_menu/start_menu.scss [hunk 5]
```diff
@@ -8,6 +8,16 @@
 .start-menu__modal-backdrop {
   z-index: 500;
 }
+.start-menu__options {
+  display: flex;
```
Comment (gdpelican) [inline angular/core/components/start_menu/start_menu.scss]: Ooh flex, nice.

File: angular/core/models/group_model.coffee (hunk 4 above)
Comment (gdpelican) [inline angular/core/models/group_model.coffee]: Yes, thanks!

File: angular/test/protractor/group_spec.coffee [hunk 6]
```diff
@@ -182,3 +280,25 @@ describe 'Group Page', ->
                  '.change-volume-form__submit')
 
       page.expectText('.group-volume-card', 'Email everything')
+
+  describe 'starting a discussion', ->
```
Comment (gdpelican) [inline angular/test/protractor/group_spec.coffee]: Great to have some robust e2ez around this stuff... will be really nice to feel like the new group form is really sorted.

**Comment by gdpelican** [comment]
Looking at that screenshot, could we draft up some placeholders to put in the name / description fields? Feels a bit 'welcome to the void' somehow. @rdbartlett @mhjb @robguthrie

**Comment by gdpelican** [comment]
This is great work, and the e2ez make it so sweet and understandable what the intention is.

**Pull request merged**